### PR TITLE
Hdfs/Zookeeper & Oracle/Scientific

### PR DESCRIPTION
**Pull request by shurl**
This adds support for the HDFS and Zookeeper plugins. 

Also adds support for Oracle and Scientific distros.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,22 +25,24 @@ This cookbook installs the v2 [Server Density](http://www.serverdensity.com/) ag
...
-#### Gems  
+#### Gems
 - `rest-client`
 
 #### Platforms:
 - Ubuntu
 - Debian
 - RHEL
 - CentOS
+- Amazon
```
Comment (carlosperello) [inline README.md]: Where is Oracle and Scientific?

File: recipes/default.rb [hunk 2]
```diff
@@ -186,6 +186,32 @@
...
+        source 'hdfs_datanode.yaml.erb'
+        owner 'sd-agent'
+        group 'sd-agent'
+        mode 0644
+        variables(
+                  :hdfs_datanode_jmx_uri => node['serverdensity']['hdfs_datanode_jmx_uri'],
+                  )
+        notifies :restart, 'service[sd-agent]', :delayed
+    end
+    if node['serverdensity']['hdfs_namenode_jmx_uri']
```
Comment (carlosperello) [inline recipes/default.rb]: Bad indentation ?

File: README.md (hunk 1 above)
Comment (shurl) [inline README.md]: Added!

**Comment by shurl** [comment]
@carlosperello I've addressed the issues you pointed out. 

The bad indentation was intentional, but after chatting yesterday I've rearranged the logic. It doesn't appear to attempt the install on the second instance when testing: 

``` 
* yum_package[sd-agent-hdfs] action install
   - install version 2.1.4-1.el7.centos of package sd-agent-hdfs
* template[/etc/sd-agent/conf.d/hdfs_datanode.yaml] action create (up to date)
* yum_package[sd-agent-hdfs] action install (up to date)
* template[/etc/sd-agent/conf.d/hdfs_namenode.yaml] action create (up to date)
* yum_package[sd-agent-varnish] action install
```

**Comment by carlosperello** [comment]
👍  will merge on Monday.

**Pull request merged**